### PR TITLE
improve #938 watch perf with search array

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: src/logic/getFieldsValues.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
-import { FieldName, FieldValues, FieldRefs } from '../types';
 import isString from '../utils/isString';
+import isArray from '../utils/isArray';
+import isUndefined from '../utils/isUndefined';
+import { FieldName, FieldValues, FieldRefs } from '../types';
 
 export default <FormValues extends FieldValues>(
   fields: FieldRefs<FormValues>,
-  search?: any,
+  search?: unknown,
```
Comment (cdagli) [inline src/logic/getFieldsValues.ts]: @bluebill1049 Possibly we have a bug here. When you use `watch({nest: true})` search becomes an object here: https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L650

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts]: good pick up, i am doing type check on fieldNames on array and string only. let me know if you actually encounter issues with it.

Comment (cdagli) [reply src/logic/getFieldsValues.ts]: Please check these 2: 
4.7.3 (works) => https://codesandbox.io/s/react-hook-form-473-next0-watchnesttrue-iw6xs
4.8.1 (fails) => https://codesandbox.io/s/react-hook-form-481-watchnesttrue-iw6xs 

image

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts]: oh. you are right thanks a lot

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts]: i will do a patch right away, you want to take a quick look the hot fix?

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts]: https://github.com/react-hook-form/react-hook-form/commit/171ab4fe88f5d088a59cf3bc0d18b2be4a8c40a1
i will follow up with unit test and autoamtion afterward. hopefully, fewer ppl get affected by this issue.

Comment (cdagli) [reply src/logic/getFieldsValues.ts]: Sorry, I missed your previous message but now it works. Thanks!

**Pull request merged**